freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -159,11 +165,34 @@ public void userSetPassword(String user, String password) throws FreeIpaClientEx
         // FreeIPA expires any password that is set by another user. Work around this by
         // performing a separate API call to set the password expiration into the future
         userMod(user, "userpassword", password);
-        updateUserPasswordExpiration(user);
+        updateUserPasswordMaxExpiration(user);
+    }
+
+    public void updateUserPasswordExpiration(String user, long expiration) throws FreeIpaClientException {
+        String passwordExpirationDate = formatDate(expiration);
+        userMod(user, "setattr", "krbPasswordExpiration=" + passwordExpirationDate);
```

Comment:
So above you set the krbPasswordExpiration with a date time string (with a certain pattern). Here it's just a long. Are both correct? If they are, let's settle on one way to identity the expiration time. Personally I like to use java Instant and Duration to hold and calculate time - it is more robust than passing along longs.

Reply:
all expirations are currently passed through the method parameters as longs. They are converted to the map {"setattr":"krbPasswordExpiration=1234567890Z"} when constructing the ipa JSON-RPC request. The userMod() method doesn't know anything about password expiration.